### PR TITLE
chore: Refactor frontend layouts

### PR DESCRIPTION
**Pull request by zurdi15**
- Frontend layout refactor
- Added UI options to customize the home page 
- Small UI tweaks
- Standarized views url with the views name

### CODE REVIEW

File: frontend/src/components/Gallery/AppBar/Collection/CollectionInfoDrawer.vue [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import type { Events } from "@/types/emitter";
+import type { Emitter } from "mitt";
+import { storeToRefs } from "pinia";
+import { inject, ref } from "vue";
+import { useDisplay } from "vuetify";
+
+// Props
+const { xs } = useDisplay();
+const emitter = inject<Emitter<Events>>("emitter");
+const viewportWidth = ref(window.innerWidth);
```
Comment (gantoine) [inline frontend/src/components/Gallery/AppBar/Collection/CollectionInfoDrawer.vue:18]: is this dynamic? ie will the component rerender when the window size changes?

File: frontend/src/components/common/Game/Card/Base.vue [hunk 2]
```diff
@@ -110,23 +117,19 @@ const computedAspectRatio = computed(() => {
...
+                  ? `/assets/default/cover/big_${theme.global.name.value}_unmatched.png`
+                  : (rom.igdb_id || rom.moby_id) && !rom.has_cover
                     ? `/assets/default/cover/big_${theme.global.name.value}_missing_cover.png`
-                    : rom.igdb_url_cover
-                      ? rom.igdb_url_cover
-                      : rom.moby_url_cover
+                    : `/assets/romm/resources/${rom.path_cover_l}?ts=${rom.updated_at}`
+                : !rom.igdb_url_cover && !rom.moby_url_cover
+                  ? `/assets/default/cover/big_${theme.global.name.value}_missing_cover.png`
+                  : rom.igdb_url_cover || rom.moby_url_cover)
```
Comment (gantoine) [inline frontend/src/components/common/Game/Card/Base.vue:132]: eventually this should be a util function that's more clearly defined/parsable

File: frontend/src/components/common/Game/Dialog/MatchRom.vue [hunk 3]
```diff
@@ -495,28 +494,3 @@ onBeforeUnmount(() => {
...
-  z-index: 9999 !important;
-}
-.source-cover-desktop {
-  min-width: 220px;
-  max-width: 220px;
-}
-.source-cover-mobile {
-  min-width: 150px;
-  max-width: 150px;
-}
```
Comment (gantoine) [inline frontend/src/components/common/Game/Dialog/MatchRom.vue:521]: 👍🏼

File: frontend/src/components/common/Platform/Dialog/DeletePlatform.vue [hunk 4]
```diff
@@ -71,7 +71,7 @@ function closeDialog() {
           >{{ platform.name }} - [<span class="text-romm-accent-1">{{
             platform.fs_slug
           }}</span
-          >] from RomM.</span
+          >].</span
```
Comment (gantoine) [inline frontend/src/components/common/Platform/Dialog/DeletePlatform.vue]: shouldnt we keep "from romm" so users know it's not getting removed from the fs?

File: frontend/src/utils/index.ts [hunk 5]
```diff
@@ -416,6 +495,12 @@ export function getTextForStatus(status: PlayingStatus) {
...
 
+/**
+ * Get the status key for a given text.
+ *
+ * @param text The text to convert.
+ * @returns The corresponding status key.
+ */
 export function getStatusKeyForText(text: string) {
   return inverseRomStatusMap[text];
 }
```
Comment (gantoine) [inline frontend/src/utils/index.ts:506]: ❤️ ❤️ ❤️

File: frontend/src/views/Scan.vue [hunk 6]
```diff
@@ -99,13 +100,14 @@ async function stopScan() {
   <v-row class="align-center pt-4 px-4" no-gutters>
     <!-- Platform selector -->
     <v-col cols="12" md="5" lg="6" class="px-1">
+      <!-- TODO: add 'ALL' default option -->
```
Comment (gantoine) [inline frontend/src/views/Scan.vue:103]: todo in this PR or future work?

**Review by gantoine — APPROVED** [review]

File: frontend/src/components/common/Game/Card/Base.vue (hunk 2 above)
Comment (zurdi15) [reply frontend/src/components/common/Game/Card/Base.vue:132]: Totally agree, too much logic here

File: frontend/src/components/common/Platform/Dialog/DeletePlatform.vue (hunk 4 above)
Comment (zurdi15) [reply frontend/src/components/common/Platform/Dialog/DeletePlatform.vue]: Hmm good point, I thought it was too repetitive, but I can readd it

File: frontend/src/views/Scan.vue (hunk 6 above)
Comment (zurdi15) [reply frontend/src/views/Scan.vue:103]: Future PR, just to not forget

File: frontend/src/components/Gallery/AppBar/Collection/CollectionInfoDrawer.vue (hunk 1 above)
Comment (zurdi15) [reply frontend/src/components/Gallery/AppBar/Collection/CollectionInfoDrawer.vue:18]: Hmmm not sure right now, as soon as I can test it again I'll check but it should be

Comment (zurdi15) [reply frontend/src/components/Gallery/AppBar/Collection/CollectionInfoDrawer.vue:18]: It is!

**Review by adamantike — APPROVED** [review]

File: frontend/src/components/common/Game/Card/Base.vue (hunk 2 above)
Comment (adamantike) [reply frontend/src/components/common/Game/Card/Base.vue:132]: Moving this to a separate function in a future PR will also simplify fixing the issue where frontend sometimes sends `?ts=undefined`

**Pull request merged**